src/cargo/core/package_id_spec.rs

```diff
@@ -181,7 +181,22 @@ impl PackageIdSpec {
         }
 
         if let Some(u) = &self.url {
-            if u != package_id.source_id().url() {
+            let package_base_url = format!(
+                "{}://{}",
+                u.scheme(),
+                u.host_str().expect("package spec url should have a host")
+            );
+            let source_id = package_id.source_id();
+            let source_id_url = source_id.url();
+            let source_id_base_url = format!(
+                "{}://{}",
+                source_id_url.scheme(),
+                source_id_url
+                    .host_str()
+                    .expect("source id url should have a host")
+            );
+            // Examine only the base URL, as the package spec URL might include a package name within its path.
+            if package_base_url != source_id_base_url {
```

Comment:
I don't think its correct to completely ignore paths within the URLs, like with git sources.

Reply:
Yeah. I found it from the failed CI. We also support file URLs.